File: swift/problem482.swift
class Solution {
    func licenseKeyFormatting(_ S: String, _ K: Int) -> String {
        var ans = String(); var counter = 0
        
        for char in S.uppercased().reversed() {
            if (char != "-") {
                if (counter == K) {
                    ans += "-"
                    counter = 0
                }
                ans += String(char)
                counter += 1
            }
        }
        
        return String(ans.reversed())
    }
}

File: python/problem482.py
class Solution:
    def licenseKeyFormatting(self, S, K):
        """
        :type S: str
        :type K: int
        :rtype: str
        """
        S=S.split('-')
        S=''.join(S).upper()[::-1]
        S='-'.join(S[i:i+K] for i in range(0, len(S), K))[::-1]
        return S
